File: platform/windows/AdGuard.FilterListManager/MarshalLogic/FilterListMetadata.cs
namespace AdGuard.FilterListManager.MarshalLogic
{
    public class FilterListMetadata
    {
        public FilterListMetadata(string title,
            string description,
            string timeUpdated,
            string version,
            string homepage,
            string license,
            string checksum,
            string url,
            int rulesCount)
        {
            this.title = title;
            this.description = description;
            this.timeUpdated = timeUpdated;
            this.version = version;
            this.homepage = homepage;
            this.license = license;
            this.checksum = checksum;
            this.url = url;
            this.rulesCount = rulesCount;
        }

        public string title { get; set; }
        public string description { get; set; }
        public string timeUpdated { get; set; }
        public string version { get; set; }
        public string homepage { get; set; }
        public string license { get; set; }
        public string checksum { get; set; }
        public string url { get; set; }
        public int rulesCount { get; set; }

        public void Deconstruct(out string title, out string description, out string timeUpdated, out string version, out string homepage, out string license, out string checksum, out string url, out int rulesCount)
        {
            title = this.title;
            description = this.description;
            timeUpdated = this.timeUpdated;
            version = this.version;
            homepage = this.homepage;
            license = this.license;
            checksum = this.checksum;
            url = this.url;
            rulesCount = this.rulesCount;
        }
    }
}

File: crates/filter-list-manager/src/manager/models/filter_list_metadata.rs
//! Metadata for a remote filter list.

/// Metadata for a remote filter list. This structure represents metadata parsed from
/// the list content and it is exposed via `fetch_filter_list_metadata`.
#[derive(Clone, Debug)]
pub struct FilterListMetadata {
    /// Title from the `! Title:` metadata field.
    pub title: String,
    /// Description from the `! Description:` metadata field.
    pub description: String,
    /// Time updated from the `! TimeUpdated:` metadata field.
    pub time_updated: String,
    /// Version from the `! Version:` metadata field.
    pub version: String,
    /// Homepage from the `! Homepage:` metadata field.
    pub homepage: String,
    /// License from the `! License:` metadata field.
    pub license: String,
    /// Checksum from the `! Checksum:` metadata field.
    pub checksum: String,
    /// URL or local path where the filter content was downloaded from.
    pub url: String,
    /// Rules count in this filter list. Simply a number of non-empty lines
    /// and does not start with a comment marker.
    pub rules_count: i32,
}
